ros2_batch_job/__main__.py

```diff
@@ -257,9 +261,15 @@ def get_args(sysargv=None):
         argv, test_args = extract_argument_group(argv, '--test-args')
     else:
         build_args, test_args = extract_argument_group(build_args, '--test-args')
+    if '--coverage-filter-packages' in argv:
+        argv, coverage_filter_packages = extract_argument_group(argv, '--coverage-filter-packages')
+    else:
+        coverage_filter_packages = None
```

Comment:
I would expect this to require some special handling like line 263 depending on the order of the various greedy arguments `--build-args`, `--test-args` and now `--coverage-filter-packages`.

Reply:
Excuse me if the answer is simple but, why if '--coverage-filter-packages' is not in argv do we need to extract any argument group? BTW I have no clue why is that done in 263 for test/build which probably explains why I can not answer your question.

Reply:
> why if '--coverage-filter-packages' is not in argv do we need to extract any argument group?

I wasn't referring to this specific line rather than to the logic to extract `--coverage-filter-packages`.

> I have no clue why is that done in 263 for test/build which probably explains why I can not answer your question.

Both existing options `--build-args` and `--test-args` are greedy and are extracted using `extract_argument_group`. Since their order is unknown the above logic makes sure that it works independent of which of the two are specified first.

With a third option the combinations in which order they appear is getting more complex (2 different orders for 2 options -> 6 different orders for 3 options).

Reply:
I added a bit more of logic in c1c28a2 although I'm not so sure that we cover all scenarios.

Reply:
I don't see how the referenced commit addressed the problem?